onnxruntime/core/framework/session_state_initializer.cc

```diff
@@ -181,37 +186,48 @@ common::Status SaveMLValueNameIndexMapping(const GraphViewer& graph_viewer,
   return Status::OK();
 }
 
-common::Status DeserializeTensorProto(const ONNX_NAMESPACE::TensorProto& tensor_proto,
-                                      const OrtAllocatorInfo& alloc_info,
-                                      const ExecutionProviders& exec_providers,
-                                      MLValue& mlvalue, void* preallocated, size_t preallocated_size) {
-  auto alloc_ptr = utils::GetAllocator(exec_providers, alloc_info);
-  if (!alloc_ptr) {
-    return Status(common::ONNXRUNTIME, common::FAIL, "Failed to get allocator for alloc_info: " + alloc_info.ToString());
-  }
-
+// This function won't check if the preallocated buffer(p_data) has enough room.
+common::Status DeserializeTensorProto(const Env& env, const std::basic_string<PATH_CHAR_TYPE>& proto_path,
```

Comment:
Can it check? If the buffer is coming from a user, and the size is wrong, don't we own ensuring there's no buffer overflow?

Reply:
This is a local function.  Only this file can use this function.

Reply:
Maybe a better comment is that 'GetTensorByTypeFromTensorProto will ensure that there's no buffer overrun, so this doesn't need to check'. I think the setup is correct as that's the best level to make the check.

Reply:
Unfortunately, it's not true.

---
In reply to: [261704126](https://github.com/Microsoft/onnxruntime/pull/520#discussion_r261704126) [](ancestors = 261704126)

Reply:
Not following what's not ture. The length is checked in GetTensorByTypeFromTensorProto isn't it?

Reply:
But if the tensor need be copied, we won't check it.